openpype/hosts/maya/plugins/create/create_layout.py

```diff
@@ -1,16 +1,19 @@
 from openpype.hosts.maya.api import plugin
+from openpype.lib import BoolDef
 
 
-class CreateLayout(plugin.Creator):
+class CreateLayout(plugin.MayaCreator):
     """A grouped package of loaded content"""
 
-    name = "layoutMain"
+    identifier = "io.openpype.creators.maya.layout"
     label = "Layout"
     family = "layout"
     icon = "cubes"
 
-    def __init__(self, *args, **kwargs):
-        super(CreateLayout, self).__init__(*args, **kwargs)
-        # enable this when you want to
-        # publish group of loaded asset
-        self.data["groupLoadedAssets"] = False
+    def get_instance_attr_defs(self):
+
+        return [
+            BoolDef("groupLoadedAssets",
+                    label="Group Loaded Assets",
```

Comment:
Good question - I have no idea what this does. Haha. It's implemented by @moonyuet with [this commit](https://github.com/ynput/OpenPype/commit/d57fdcf7971bd99422b676a2a9074b2f7eb4e767) in this PR https://github.com/ynput/OpenPype/pull/4098

I'll add least add the original comment message as tooltip which was:

> enable this when you want to publish group of loaded asset

Admittedly it's still weird even with that, but maybe I know too little about the layout publishing system for it to make sense to me.

Reply:
Yeah it's a bit too generic so @moonyuet would be great if you could add some more context to it

Reply:
> Good question - I have no idea what this does. Haha. It's implemented by @moonyuet with [this commit](https://github.com/ynput/OpenPype/commit/d57fdcf7971bd99422b676a2a9074b2f7eb4e767) in this PR #4098
> 
> I'll add least add the original comment message as tooltip which was:
> 
> > enable this when you want to publish group of loaded asset
> 
> Admittedly it's still weird even with that, but maybe I know too little about the layout publishing system for it to make sense to me.

the layout creator(and its extractor) only works for the published contents from the openpype loaders, and if it is the raw contents or any published content, it will give the assertion errors.